### ISSUE TITLE
Why does the window pop up again when I use nuitka to package the gui?

### ISSUE THREAD

**Issue by Lishumuzixin** [ultralytics/ultralytics#20279]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Background:I wrote an interface using tkinter. There was a button on the interface. After clicking it, the model was exported. However, when I clicked it, a new interface would pop up again. Why is that?
The code is as follows:
1.main.py

```
import ttkbootstrap as tb
from tkinter import filedialog
from PIL import Image, ImageTk
from ultrain.seg_export import YoloModelExport
from tkinter import messagebox
import multiprocessing as mp
import sys
import os
current_dir = os.path.dirname(os.path.abspath(__file__))#避免打包时识别不了tool目录
sys.path.append(current_dir)

def ModelExport(pt_data_path,tab_train):
    if pt_data_path == '':
        messagebox.showwarning("警告", "请选择需要导出的模型文件！")
    else:
        process = mp.Process(target=YoloModelExport, args=(pt_data_path,))
        process.start()
        while process.is_alive():
            tab_control.update()

def DisableInput(entry_text,folder):
    entry_text.config(state="normal")  # 设置输入框为可编辑状态
    entry_text.delete(0, tb.END)  # 清空切分库的Entry显示
    entry_text.insert(0, folder)  # 将结果放入切分库的Entry
    entry_text.config(state="readonly")  # 设置输入框为只读状态

def BrowseOriginalFile(file_location,defect_label_text=None):
    global defect_num #增强的缺陷类别

    defect_file_path = filedialog.askopenfilename() # defect_file_path为选择的文件路径
    DisableInput(file_location, defect_file_path)

def CreatModelTrain(tab_train):
    frame5 = tb.Frame(tab_train, width=800, height=50)
    frame5.grid(row=0, column=0,sticky="ew")
    label5 = tb.Label(frame5, text='模型导出:', width=8, bootstyle="dark")
    label5.grid(row=0, column=0, padx=(160, 40), pady=(10, 0))
    folder_icon5 = Image.open(os.path.join(os.path.dirname(__file__), 'openfile.png'))  # 替换为实际的文件夹图标文件
    folder_icon5 = folder_icon5.resize((18, 20))
    folder_icon5 = ImageTk.PhotoImage(folder_icon5)  # 将图像加载到python中
    folder_button5 = tb.Button(frame5, image=folder_icon5,
                               command=lambda: BrowseOriginalFile(file_location5),
                               bootstyle="link")
    folder_button5.image = folder_icon5  # 需要保持对图标的引用，否则图标不会显示
    folder_button5.grid(row=0, column=2, pady=(10, 0), padx=(0, 50))
    file_location5 = tb.Entry(frame5, width=90, foreground='#000000', state="readonly")
    file_location5.grid(row=0, column=1, pady=(10, 0))
    predict_button = tb.Button(frame5, text="导出", width=10,
                               command=lambda: ModelExport(file_location5.get(), tab_train),
                               style="dark.Outline.TButton")
    predict_button.grid(row=0, column=3, pady=(10, 0))

def CreatTab(win):
    global tab_control  #选项卡控件
    style = tb.Style()
    style.configure("TNotebook.Tab")

    tab_control = tb.Notebook(win, bootstyle="light")
    tab_train = tb.Frame(tab_control)
    tab_control.add(tab_train, text='模型训练')
    tab_control.pack(expand=1, fill='both')

    CreatModelTrain(tab_train)

def CreatWindow():
    global main_win 
    main_win = tb.Window(themename='morph')
    main_win.title("ABC")
    main_win.geometry("1320x850")
    main_win.resizable(False,False)
    #创建四个选项卡
    CreatTab(main_win)
    main_win.mainloop()


if __name__ == "__main__":
    global possible_encodings   #json文件中可能编码
    possible_encodings = ['utf-8', 'ANSI', 'utf-8-sig']
    mp.freeze_support()
    split_win=None
    CreatWindow()
```

2.seg_export.py
```
def YoloModelExport(pt_path):
    print("\n-------- 开始导出模型 --------")
    model = YOLO(pt_path)  # load a custom trained
    model.fuse()
    model.info(verbose=False)  # Print model information
    model.export(format="onnx",opset=12,simplify=True,dynamic=True)
```

The packaging command is:
`nuitka --msvc=latest --standalone --show-progress --onefile --nofollow-import-to=scipy,searbon,ultralytics,torchvision,PIL,difflib,cpuinfo,tqdm,pandas,xml,yaml,pytz,ttkbootstrap,mpmath,xpinyin,onnxsim,rich,onnx,torch,encodings,google,cv2,sympy --include-package=logging,ctypes --include-module=psutil --include-package-data=ultrain  --include-data-file=./openfile.png=./   --include-data-file=./ultrain/yolov8-seg.yaml=ultrain/yolov8-seg.yaml --enable-plugins=tk-inter,numpy,matplotlib --output-dir=output main.py`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Lishumuzixin, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response. An Ultralytics engineer will review your issue and assist you as soon as possible. Thank you for your patience! 😊

**Comment by Lishumuzixin** [comment]
Besides, this seems to be related to YOLO's export. Is there anything in this part of the code that will restart the new process?

**Comment by glenn-jocher** [comment]
The issue is happening because when you run `model.export()` in your `YoloModelExport` function, the Ultralytics export process is likely creating its own window. This is especially apparent when using multiprocessing since you're starting a completely new Python process.

Try modifying your export function to disable any visualizations by setting `verbose=False`:

```python
def YoloModelExport(pt_path):
    print("\n-------- 开始导出模型 --------")
    model = YOLO(pt_path)
    model.fuse()
    # Add verbose=False to export to prevent window from appearing
    model.export(format="onnx", opset=12, simplify=True, dynamic=True, verbose=False)
```

Also, in your Nuitka packaging command, I notice you're using `--nofollow-import-to=ultralytics` which could cause issues. Since your application depends on the ultralytics library for model export, you should include it properly in your packaged application rather than excluding it.

**Comment by Lishumuzixin** [comment]
Hello, what you said about setting verbose=False is of no use. The use of --nofollow-import-to=ultralytics is because direct packaging always reports an error. My approach is to first eliminate ultralytics and then package it, and then put this package into.dist.

**Comment by glenn-jocher** [comment]
Thanks for providing more details. Looking deeper at the exporter code, I believe this might be related to the callbacks system in Ultralytics' export process.

Try this solution:

```python
def YoloModelExport(pt_path):
    print("\n-------- 开始导出模型 --------")
    # Change how you import to control the environment
    import os
    os.environ["ULTRALYTICS_MODE"] = "headless"  # Try to prevent UI elements
    
    from ultralytics import YOLO
    model = YOLO(pt_path)
    model.fuse()
    
    # Clear any callbacks that might create windows
    model.reset_callbacks()
    model.export(format="onnx", opset=12, simplify=True, dynamic=True, verbose=False, device="cpu")
```

For your Nuitka packaging approach, that's understandable. An alternative approach would be to create a custom exporter that directly uses the required components from Ultralytics without triggering the standard export pipeline. However, that would require more significant code changes.
